docs/binary-transparency-for-artifact-registries.md

```diff
@@ -0,0 +1,160 @@
+
+BT enhances the security posture of PEP 458\. Without BT, PEP 458 requires trust in the registry. Certain attacks are mitigated with respect to the distribution of artifacts, but the registry can still maliciously serve a new artifact to users without detection. There is an element of discoverability since metadata must be committed cryptographically and served via a TUF repository, but this metadata is signed by the registry. This makes PEP 458's threat model similar to registry-signed artifacts where there is still trust in the registry. In addition, the registry can choose to provide split-views to users, where the registry serves different TUF metadata to different callers. BT mitigates this threat by requiring that a log's state be independently verified, or witnessed, to prevent split-view attacks.
+
+PEP 458 and TUF are also complementary to BT. BT only mitigates the threat of a package identifier mapping to multiple hashes, but requires a fully resolved package identifier, such as `name.platform@version`. TUF can provide a mapping between a partial identifier, such as `name@version`, and the fully resolved identifier, where that identifier is then verified to be included in a log.
+
+## Design Ideas
+
+Detailed design will be fleshed out in a later document.
+
+### High-Level Requirements
+
+* An easy-to-deploy-and-manage binary transparency log. Current examples include [Tessera](https://github.com/transparency-dev/trillian-tessera), [Serverless](https://github.com/transparency-dev/serverless-log) and [Sunlight](https://github.com/FiloSottile/sunlight) (designed for Certificate Transparency)  
+* Registry support for uploading package identifiers mapped to artifact hashes to a log and providing off-line verifiable proofs of inclusion.  
+* Tooling integrated into package managers to seamlessly and transparently verify proofs of inclusion without any configuration from the user.
+
+### Deployment
+
+* Binary Transparency must incorporate all elements of a complete transparency ecosystem integration, with strong offline proofs that bundle an inclusion proof and co-signed witness checkpoints. Witnesses will verify the consistency of the log, that the log remains append-only, as is the requirement for any transparency ecosystem.  
+* A log must restrict publishing to only the registry operator. Otherwise, it is not possible to monitor the log for unexpected artifact hashes and hold the registry accountable. A registry may choose to operate the log itself, but the log could also be operated by another entity, as long as in either case, log publication is restricted to the registry operator.
+
+## Claimant Model
+
+Terminology defined in [Trillian documentation](https://github.com/google/trillian/tree/master/docs/claimantmodel).
+
+* The Claim is "There exists at most one mapping between a given artifact's package identifier and its cryptographic hash".  
```

Comment:
This claim doesn't encode the filename.

Reply:
Does "package identifier" point at the triple {module, version, filename} ?

Reply:
I see that the top of the doc says that this triple is the package identifier. Closing this comment.